### ISSUE TITLE
package:discover Filesystem Permissions of Output changed in 5.7

### ISSUE THREAD

**Issue by h2513289** [laravel/framework#26287]
- Laravel Version: 5.7.10
- PHP Version: 7.2.11
- Database Driver & Version: Not relevant

### Description:
In Laravel 5.7 the PackageManifest::write function was changed from Filesystem::put to FIlesystem::replace. 
#### 5.6 
Filesystem::put executes a file_put_contents to bootstrap/cache/packages.json, keeping permissions and ownership intact
#### 5.7
Filesystem::replace creates a temp file using tempnam, then renames it to bootstrap/cache/packages.json. 
From http://php.net/manual/en/function.tempnam.php : "Creates a file with a unique filename, with access permission set to 0600, in the specified directory."

Thus, if "composer install" or "composer update" are invoked by a different user than the one running the webserver (typically a dedicated account), bootstrap/cache/packages.json is not readable by the webserver causing every page load to error out

### Steps To Reproduce:
```bash
chmod 644 bootstrap/cache/*.php
composer install
ls -al bootstrap/cache/
```

**Comment by mfn** [comment]
This was fixed in 5.7.11 already (as in: it was reverted)

**Comment by h2513289** [comment]
Awesome, thanks for the info @mfn !

https://github.com/laravel/framework/blob/5.7/CHANGELOG-5.7.md#v5711-2018-10-24 has the relevant entry.
